packages/oauth-helper/src/index.ts

```diff
@@ -0,0 +1,468 @@
+    this.installOptions = options;
+
+    // Base URL
+    let URL = 'https://slack.com/oauth/v2/authorize?';
+    if (this.authVersion === 'v1') {
+      URL = 'https://slack.com/oauth/authorize?';
+    }
+
+    // scope
+    let scopes;
+    if (options.scopes instanceof Array) {
+      scopes = options.scopes.join(',');
+    } else {
+      scopes = options.scopes;
+    }
+    URL = `${URL}scope=${scopes}`;
+
+    // generate state
+    const state = await this.stateStore.generateStateParam(new Date(), options.metadata);
+    URL = `${URL}&state=${state}`;
+
+    // client id
+    URL = `${URL}&client_id=${this.clientId}`;
+
+    // redirect uri
+    if (options.redirectUri != null) {
+      URL = `${URL}&redirect_uri=${options.redirectUri}`;
+    }
+
+    // team id
+    if (options.teamId != null) {
+      URL = `${URL}&team=${options.teamId}`;
+    }
+
+    // user scope, only available for OAuth v2
+    if (options.userScopes != null && this.authVersion === 'v2') {
+      let userScopes;
+      if (options.userScopes instanceof Array) {
+        userScopes = options.userScopes.join(',');
+      } else {
+        userScopes = options.userScopes;
+      }
+      URL = `${URL}&user_scope=${userScopes}`;
+    }
+
+    return new Promise<string>((resolve) => {
+      resolve(URL);
+    });
+  }
+
+  /**
+   * This method handles the incoming request to the callback URL.
+   * It can be used as a RequestListener in almost any HTTP server
+   * framework.
+   *
+   * Verifies the state using the stateStore, exchanges the grant in the
+   * query params for an access token, and stores token and associated data
+   * in the installationStore.
+   */
+  public async handleCallback(
+    req: IncomingMessage,
+    res: ServerResponse,
+    options?: CallbackOptions,
+  ): Promise<void> {
+    console.log('handleCallback');
+    console.log(options);
+    let parsedUrl;
+    let code: string = '';
+    let state: string = '';
+
+    if (req.url != null) {
+      parsedUrl = url.parse(req.url, true);
+      code = parsedUrl.query.code as string;
+      state = parsedUrl.query.state as string;
+    }
+    console.log('code', code);
+    console.log('state', state);
+
+    try {
+      const metadata = await this.stateStore.verifyStateParam(new Date(), state);
+      console.log(metadata);
+
+      const client = new WebClient('');
+
+      let resp;
+      let installation: Installation;
+      console.log('authVersion', this.authVersion);
+      if (this.authVersion === 'v1') {
+        // convert response type from WebApiCallResult to OAuthResponse
+        resp = await client.oauth.access({
+          code,
+          client_id: this.clientId,
+          client_secret: this.clientSecret,
+        }) as unknown as OAuthV1Response;
+
+        // TODO: need to get appID
+        // resp obj for v1 - https://api.slack.com/methods/oauth.access#response
+        // TODO: look into getting bot.id added to platform response
```

Comment:
`bot.id` didn't exist in the response from either `oauth.access` or `oauth.v2.access`

Reply:
Great suggestion. I can get the bot.id with this. @aoberoi do you think it is beneficial to make this extra api call (no additional scopes required) to get the `bot.id`?

Reply:
for `v1`, it turns out `auth.test` only returns `bot_id` if you use your `xoxb` token when making the call. Problem is, I don't have a `xoxb` token when doing `v1` call. I tried to fix this by adding a `bot user` to my application, but it didn't solve the problem. I discussed this at https://github.com/slackapi/node-slack-sdk/pull/963/files#r379130671

Reply:
Resolved. needed to add the `bot` scope to my request